library/blocks/in-world-recipes/src/main/java/org/quiltmc/qsl/inworldrecipes/api/InWorldRecipe.java

```diff
@@ -4,19 +4,26 @@
 import net.minecraft.item.ItemUsageContext;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.Set;
+
 /**
  * Represents an <em>in-world recipe</em> - a recipe that can be performed on a specific block type in the world.
  */
 public interface InWorldRecipe {
 	/**
-	 * {@return the specific block type required for this recipe}
+	 * {@return the specific set of block types required for this recipe}
 	 */
-	@NotNull Block targetBlock();
+	@NotNull Set<Block> targetBlocks();
 
 	/**
 	 * Tries to perform the recipe.
 	 * @param context item usage context
 	 * @return {@code true} if recipe was successfully performed, {@code false} otherwise
 	 */
 	boolean tryPerform(@NotNull ItemUsageContext context);
+
+	/**
+	 * Invoked on data pack reload. Use this to update your {@linkplain #targetBlocks() target block set}.
+	 */
+	default void onReload() { }
```

Comment:
Is this needed? You could just recalculate the set every time in `targetBlocks`, considering it'll only be used on datapack reload to recreate the multimap.

Reply:
Fair point, method yeeted.